src/service/use-service.ts

```diff
@@ -0,0 +1,676 @@
+import { Application, NullableId } from '@feathersjs/feathers'
+import { ref, computed, unref, set, del, Ref } from 'vue-demi'
+import { MaybeArray, MaybeRef } from '../utility-types'
+
+import type { HandleEvents, Params } from '../types'
+import type { Id } from '@feathersjs/feathers'
+import sift from 'sift'
+import { operations } from '../utils-custom-operators'
+import { _ } from '@feathersjs/commons'
+import { filterQuery, sorter, select } from '@feathersjs/adapter-commons'
+import fastCopy from 'fast-copy'
+import {
+  getId,
+  getTempId,
+  getAnyId,
+  getQueryInfo,
+  assignTempId,
+  cleanData,
+  restoreTempIds,
+  getArray,
+  hasOwn,
+  getSaveParams,
+} from '../utils'
+
+import type {
+  AnyDataOrArray,
+  HandleFindResponseOptions,
+  AnyData,
+  UseFindWatchedOptions,
+  UseGetOptions,
+  FindClassParams,
+  FindClassParamsStandalone,
+  GetClassParams,
+  GetClassParamsStandalone,
+  ModelConstructor,
+} from './types'
+
+import { useServiceEvents } from './use-service-events'
+import { useFind as _useFind } from '../use-find'
+import { useGet as _useGet } from '../use-get'
+import { useFindWatched as _useFindWatched } from '../use-find-watched'
+import { useGetWatched as _useGetWatched } from '../use-get-watched'
+import { defaultIdField, defaultTempIdField } from './utils'
+import { BaseModel } from './base-model'
+import { useServiceClones } from './use-service-clones'
+import { useServicePending } from './use-service-pending'
+import { useServicePagination } from './use-service-pagination'
+
+export type UseFeathersServiceOptions<C extends ModelConstructor = ModelConstructor, M = InstanceType<C>> = {
+  app: Application
+  Model?: C
+  idField?: string
+  tempIdField?: string
+  clientAlias?: string
+  servicePath: string
+  whitelist?: string[]
+  paramsForServer?: string[]
+  skipRequestIfExists?: boolean
+  ssr?: MaybeRef<boolean>
+  handleEvents?: HandleEvents<C>
+  debounceEventsTime?: number
+  debounceEventsGuarantee?: boolean
+}
+
+const defaultSharedState = {
+  clientAlias: 'api',
+  skipRequestIfExists: false,
+}
+
+const FILTERS = ['$sort', '$limit', '$skip', '$select']
+const additionalOperators = ['$elemMatch']
+
+export const useService = <C extends ModelConstructor = ModelConstructor, M = InstanceType<C>>(
+  _options: UseFeathersServiceOptions<C, M>,
+) => {
+  const options = Object.assign({}, defaultSharedState, _options)
+
+  let _Model: C
+
+  // If no Model class is provided, create a dynamic one.
+  if (!_options.Model) {
+    // @ts-expect-error - We are creating a dynamic class
+    _Model = class DynamicBaseModel extends BaseModel {
+      static dynamicBaseModel = true
+      static modelName = _options.servicePath
+    }
+  } else {
+    _Model = _options.Model
+  }
+
+  if (!_Model.modelName) {
+    _Model.modelName = _Model.name
+  }
+
+  const servicePath = computed(() => _options.servicePath)
```

Comment:
`servicePath` should probably remain a `ref`.  Part of the intention is to be able to use a single store as a sort of dynamic CRUD manager for any service, just by changing the servicePath you can manage a different set of resources.

Reply:
got ya. Thanks for the review. Reverted it.